pytest_html/plugin.py

```diff
@@ -328,16 +312,40 @@ def generate_summary_item(self):
 
         doc = html.html(head, body)
 
-        logfile.write('<!DOCTYPE html>')
         unicode_doc = doc.unicode(indent=2)
         if PY3:
             # Fix encoding issues, e.g. with surrogates
             unicode_doc = unicode_doc.encode('utf-8',
                                              errors='xmlcharrefreplace')
             unicode_doc = unicode_doc.decode('utf-8')
-        logfile.write(unicode_doc)
+        return unicode_doc
+
+    def _save_report(self, report_content):
+        if not os.path.exists(os.path.dirname(self.logfile)):
+            os.makedirs(os.path.dirname(self.logfile))
+        logfile = open(self.logfile, 'w', encoding='utf-8')
+
+        logfile.write('<!DOCTYPE html>')
```

Comment:
This means that `_generate_report` returns a report that doesn't include this doctype declaration. I think it makes sense to include this in `_generate_report`. Otherwise, I'm happy with this change.